lib/cucumber/runtime.rb

```diff
@@ -229,6 +229,7 @@ def filters
         filters << Cucumber::Core::Test::LocationsFilter.new(filespecs.locations)
         filters << Filters::Randomizer.new(@configuration.seed) if @configuration.randomize?
         filters << Filters::Quit.new
+        filters << Filters::Retry.new(@configuration)
```

Comment:
I expect the following behaviour:
```
  Given I use retry
  And a long-running, failing test has started to execute
  When Cucumber.wants_to_quit is set
  Then the long-running, failing test case is only executed once 
```
As far as I can tell, the order of the `Quit` and the `Retry` filter needs to be swapped, to get this behaviour.